base/inference.jl

```diff
@@ -542,7 +542,38 @@ add_tfunc(===, 2, 2,
         end
         return Bool
     end)
-add_tfunc(isdefined, 1, IInf, (args...)->Bool)
+function isdefined_tfunc(args...)
+    arg1 = args[1]
+    if isa(arg1, Const)
+        a1 = typeof(arg1.val)
+    else
+        a1 = widenconst(arg1)
+    end
+    if isType(a1)
+        return Bool
+    end
+    if isleaftype(a1)
```

Comment:
Also can do `a1 = unwrap_unionall(a1)` first (before the `isType`) check as well.